lib/db.js

```diff
@@ -238,6 +238,12 @@ module.exports = (
 
     return P.all(promises)
       .spread((mysqlSessionTokens, redisTokens) => {
+        if (config.tokenLifetimes.sessionTokenWithoutDevice) {
+          // Filter out any expired sessions
+          mysqlSessionTokens = mysqlSessionTokens.filter(sessionToken =>
+            sessionToken.createdAt > Date.now() - config.tokenLifetimes.sessionTokenWithoutDevice
```

Comment:
Oh, whoops. I stupidly assumed that the `sessions` stored procedure only returns web sessions, but I just checked the code and that's not true. A remote test would have caught this of course.